client-common/src/tendermint/types/block_results.rs

```diff
@@ -49,6 +53,25 @@ impl BlockResults for BlockResultsResponse {
         }
     }
 
+    fn contains_staking(&self) -> bool {
+        match &self.txs_results {
```

Comment:
how about exit early, if self.txs_results is Noe, and remove `match`

Reply:
yeah, I use `if let Some(x)...`, it can reduce code.